Kernel/VM/MemoryManager.cpp

```diff
@@ -811,15 +808,12 @@ void MemoryManager::deallocate_physical_page(PhysicalAddress paddr)
     }
 
     // If it's not a user page, it should be a supervisor page.
-    for (auto& region : m_super_physical_regions) {
-        if (!region.contains(paddr)) {
-            dbgln("MM: deallocate_supervisor_physical_page: {} not in {} - {}", paddr, region.lower(), region.upper());
-            continue;
-        }
-
-        region.return_page(paddr);
+    if (m_super_physical_region->contains(paddr)) {
+        m_super_physical_region->return_page(paddr);
         --m_system_memory_info.super_physical_pages_used;
         return;
+    } else {
+        dbgln("MM: deallocate_supervisor_physical_page: {} not in {} - {}", paddr, m_super_physical_region->lower(), m_super_physical_region->upper());
```

Comment:
Maybe we should handle the `!contains()` case first. That way we could get rid of the else branch and the return.

Reply:
right but we dont want to return, we want to panic, i can still convert it to `!contains` and move the panic inside, but that feels a bit weird, since the panic is not specific to this check.